### ISSUE TITLE
UNBLOCKING INSTRUCTIONS WITH PROOFANDROID AUTO 3.0 BLOCKING INSTRUCTIONS WITH PROOF 

### ISSUE THREAD

**Issue by shmykelsa** [thekirankumar/carstream-android-auto#116]
http://agileurbia.com/GooT

You can thank me later

**Comment by kaisertgc** [comment]
What is this?

**Comment by shmykelsa** [comment]
Instructions I made for installing custom apps in Android Auto 3.0

**Comment by magna413** [comment]
Confirmed! 
Works!

Tried other apps too and they all work!

**Comment by kaisertgc** [comment]
It is just showing advertisement. Am I missing sth here?

**Comment by shmykelsa** [comment]
Wait 5 seconds and then on the upper side, on the right, you can skip the ads

**Comment by puzzitta** [comment]
Not working for me. Help me thank yoy

**Comment by ahhl** [comment]
shmykelsa. your instruction need to mentioned existing  AA that work with AA2.9, need to be uninstall and reinstall using the mentioned method after AA3.0 installed. this from my experience

**Comment by shmykelsa** [comment]
@puzzitta what's wrong?

@ahhl that's a good point

**Comment by puzzitta** [comment]
I'm in the ball, can you please indicate step by step how to do? Thanks so much.

**Comment by shmykelsa** [comment]
@puzzitta the steps are described in the pastebin text

**Comment by puzzitta** [comment]
Sorry shmykelsa, I followed your guide but I did not understand if I have to install AA 2.9 or the latest version of AA 3.0, because in both cases CARSTREAM does not work. Thank you very much for your availability. Good day

**Comment by shmykelsa** [comment]
@puzzitta have you uninstalled CarStream before doing the whole thing?

**Comment by puzzitta** [comment]
I uninstalled Carstream and AA 3.0, after I reinstalled AA 3.0 and then I did the Pastebin procedure. Thanks always. Greetings.

**Comment by fmagarreiro** [comment]
Hello. 
I'm trying to follow your instructions. Bur I am not able to get success. When I issue adb command in PC, the answer is:  "Error: failed to write; /SDcard/APKS/youtube_auto_v1.8.3.apk (No such file or directory)". My Android is not rooted.
Thanks in advance

**Comment by shmykelsa** [comment]
@puzzitta was it successfull?
@fmagarreiro Have you actually changed "/SDcard/APKS/youtube_auto_v1.8.3.apk" with the path your own APK is?

**Comment by fmagarreiro** [comment]
@shmykelsa thanks for your answer. 
"/SDcard/APKS/youtube_auto_v1.8.3.apk" is the actual location of the apk file to be installed.

Will help alot if you explain the meaning of each sentence in the command:

adb shell = 
pm install = 
-i = 
"com.android.vending" =
-r =
*path of the APK* = 

Thanks again

**Comment by shmykelsa** [comment]
@fmagarreiro there's ADB documentation for that

**Comment by fmagarreiro** [comment]
@shmykelsa Sorry if I'm making you loose your time with me... I am a begginer on this matters, and I was trying to understand...
Thanks anyway.

**Comment by fortelux** [comment]
![image](https://user-images.githubusercontent.com/36641644/36685956-3382525a-1b25-11e8-94a5-1cf46e730f9b.png)
@shmykelsa sorry, where is the issue?

**Comment by shmykelsa** [comment]
@fortelux you're not specifying the extension of the file. Append .apk at the end

**Comment by fortelux** [comment]
@shmykelsa Grazie!

**Comment by fortelux** [comment]
@shmykelsa sorry, other questino, after this, do i have to clone the app and unistall the older? or not?

**Comment by shmykelsa** [comment]
@fortelux no!
 ;)

**Comment by puzzitta** [comment]
I tried again but nothing does. How can I do????

**Comment by fmagarreiro** [comment]
@shmykelsa SUCESS !! Carstream it's working under  AA 3.0.

**Comment by fmagarreiro** [comment]
@magna413 what other apps did you tried ?

**Comment by shmykelsa** [comment]
@puzzitta what's wrong?

**Comment by chokito78** [comment]
Followed the steps sucesfull, but i will make sure about working just on this friday, my car is in the mechanic shop.
as soon as possible i will post here if worked ou not.
Thanks!

**Comment by puzzitta** [comment]
![36578049-ade12fd8-1839-11e8-84e8-66f30b2df49c](https://user-images.githubusercontent.com/36827398/36700524-41060ed2-1b50-11e8-9034-c15a6632029b.png)
This my error

**Comment by chokito78** [comment]
Puzzitta, CarStream just works on multimedia center on your car, not on your cell phone.

**Comment by feelfreetoblameme** [comment]
Thanks for these instructions! Worked like a charm. Just to help others, I did not uninstall Android Auto version 3, only CarStream. I also did not change the name of Kiran's APK at any point, and now it comes up as an option under the last icon in my car. There was an option in Android Auto's developer options that I'm not sure was there before, where you can change the app mode from one of five options as shown in the screenshot. I picked Developer but have no idea if this made any difference.
![screenshot_2018-02-26-15-05-40](https://user-images.githubusercontent.com/8293751/36701049-ab342248-1b06-11e8-8ecd-f14bf9602f89.png)

**Comment by shmykelsa** [comment]
@feelfreetoblameme yes it was there before, it just changes how app appears on car, and it will tell you some detailed stats on last panel. And no, it does not make the difference

**Comment by pdario** [comment]
This basically re-install the APK file without wiping its data (-r switch) as if it was installed from the Play Store (-i switch)

**Comment by kaisertgc** [comment]
Thanks much @shmykelsa .Works. What about to update the apps in future? Just to repeat this with new apks?

**Comment by shmykelsa** [comment]
@kaisertgc yes

**Comment by nameor123** [comment]
I uninstall android auto and car stream. Invoke usb debugging developer mode on android and also allow unknown installation in android auto. I was able to install and and I use the command and shell pm install -i "com.android.vending" - ... th e path file. Successful. I try to install carstream from that file and it would not show up in my car other than carstream player. Help

**Comment by shmykelsa** [comment]
@nameor123 after the command, you musn't do anything else. You must not install it with that file.

**Comment by nameor123** [comment]
@shmykelsa yeah I did install it afterwards I had a feeling shouldn't install it after that. I'll try it

**Comment by shmykelsa** [comment]
@nameor123 it basically screws up the whole thing

**Comment by nameor123** [comment]
@shmykelsa Works fine.

**Comment by lluisgener** [comment]
@shmykelsa I confirm it works, AA 3, CarStream 1.8.3, Hyundai Ioniq.

Thanks.

**Comment by mikemaxe** [comment]
thanks, it worked for me on my Seat Ateca

**Comment by Inkathu** [comment]
I've checked this solution - everything works fine!

**Comment by JHawk88** [comment]
The link provided downloads an app that posts an advertisement pretty regularly. Sounds like a hack. Thanks for nothing

**Comment by shmykelsa** [comment]
@JHawk88 do you even know how the internet works?

**Comment by JHawk88** [comment]
@shmykelsa Yes.  People give you links to websites telling you it does something that end up trying to force dls onto your stuff.  You did this exact thing with 0 instructions.  Turns out, the apps that were trying to force themselves onto my phone were not at all related.  Wait for the "skip ad in the top right..."  Still got 2 apps on my phone that, thankfully, were easy to remove.  While I def appreciate your help, sending us to a questionable website, not cool.

**Comment by lluisgener** [comment]
You must skip the ad (wait for the countdown to finish), and there it goes the info. It would've been better to post instructions here instead of a link, but the info is there, and it works.

**Comment by pythonhockey** [comment]
https://pastebin.com/q4Hicev1

The link above is the direct pastebin do avoid the ad spamminess.

**Comment by shmykelsa** [comment]
@JHawk88 lot of people succeded but not you. Cleverness?



Anyway I'm about to release a one-click tool to make it automatic. I'm currently dealing with false-positive in Windows Defender, and submitted the file to Windows Defender Security Intelligence. Maybe in next definition it will be solved.

Anyone interested in testing the one-click exe could conctact me via mail (it's on my profile)

**Comment by kaisertgc** [comment]
@shmykelsa This your solution will work on future android versions? If so your exe would be very nice easy way to do it. Thanks again.

**Comment by shmykelsa** [comment]
@kaisertgc Yes it will. It will also have ADB bundled, so just one file to run

**Comment by shmykelsa** [comment]
https://goo.gl/NAtKRN

So this is the .exe I've made to patch apks for Android Auto 3.0. It requires Windows

As always, skip ads (please note that I'm basically giving up some free time to doing this, and paying me a beer through sites that pay for ads would be nice)

Unfortunately, most of anti-viruses flag this as a virus, but it's totally safe to run, and it does not contain viruses. I'm currently working with virustotal to get this sorted out.

It's very simple to use and it does not require any installation. Just follow instructions.

Let me know if this works well with you. It's basically a CMD script compiled in exe

**Comment by mystang82** [comment]
Help me! Please
![image](https://user-images.githubusercontent.com/37029174/36940131-287962ea-1f4e-11e8-9714-f577ca70fb35.png)

**Comment by mystang82** [comment]
show video (upload to the website), show more detailed instructions, Russian users will be immensely grateful!!!

**Comment by shmykelsa** [comment]
@mystang82 You've put an invalid path. /sdcard1/ is not a drive. You might wanna check out my new windows program

**Comment by fortelux** [comment]
Hi @shmykelsa , your windows program link is not available, probably banned by mediafire....Thnx.

**Comment by shmykelsa** [comment]
@fortelux working here, but since I uploaded a new version is now another one

**Comment by fortelux** [comment]
@shmykelsa i just tried the new file .exe that u uploaded but unfortunately i think it doesn't work for me...At the request "press any key to continue"..the program exits and closes by its own.

**Comment by Vadgr** [comment]
How to open links? Manual instructions: http://agileurbia.com/GooT
the browser writes that the timeout has expired, an error.

**Comment by Vadgr** [comment]
@shmykelsa How to open links? the browser writes that the timeout has expired, an error.

**Comment by shmykelsa** [comment]
@Vadgr are you able to open this? http://activeterium.com/HUzU

**Comment by yuri761** [comment]
@shmykelsa a me non funziona

![cattura](https://user-images.githubusercontent.com/37051105/36950788-53ef1d7c-1ffb-11e8-99b6-e005d9ee140e.PNG)
@shmykelsa

**Comment by shmykelsa** [comment]
@yuri761 pensavo di aver sistemato il link. Ora sistemo. Ne ero al corrente.

**Comment by yuri761** [comment]
@shmykelsa.allora premetto che non ho usato il tuo programma.in modalità applicazione ho messo sviluppatore e funziona

**Comment by yuri761** [comment]
@shmykelsa.allora premetto che non ho usato il tuo programma.in modalità applicazione ho messo sviluppatore e funziona

**Comment by kaisertgc** [comment]
.exe file deleted from mediafire

**Comment by shmykelsa** [comment]
@kaisertgc I don't know what you're talking about. It's still up.

**Comment by kaisertgc** [comment]
@shmykelsa This link  http://activeterium.com/Ey88 directs me here. Sth am I doin wrong?


![screnshot](https://user-images.githubusercontent.com/25952363/37091952-0911dffc-221b-11e8-92bd-2c1a9c5aac15.jpg)

**Comment by shmykelsa** [comment]
Mediafire flagged my exe as a virus. I'll try to mirror it asap

**Comment by shmykelsa** [comment]
File is mirrored to another site and now up again.

**Comment by matisaul** [comment]
Not Working for me... I made every steps, but I get the same message: "Click last icon...". I'm using lastest Android Auto app from Play store (3.0.57...), and last Carstream 1.8.3 
I unistalled both programs, but still no luck with this. Any tips?

**Comment by shmykelsa** [comment]
@matisaul are you using it on phone or on the car?

**Comment by matisaul** [comment]
@shmykelsa on the car.

**Comment by shmykelsa** [comment]
@matisaul can you see it on last tab? (not the one with headphones)

**Comment by matisaul** [comment]
@shmykelsa Sorry, my bad... I forget to watch the last tab... It's working perfectly!! Thanks!

**Comment by supercharger619** [comment]
Hey guys I’ve uninstalled carstream and tried with AA2.8 and AA 3.0 with no success, I keep getting the message below. Please advise.

C:\Users\super\AppData\Local\Temp\afolder\adb.exe
Access is denied.
C:\Users\super\AppData\Local\Temp\afolder\AdbWinApi.dll
Access is denied.
C:\Users\super\AppData\Local\Temp\afolder\AdbWinUsbApi.dll
Access is denied.
Please connect your phone via USB. Continue if already connected.
Press any key to continue . . .
Please activate USB debug on your phone. Continue if already activated.
Press any key to continue . . .
Please place on desktop the APK you wanna patch and rename it “my.apk”
IF THIS APP IS ALREADY INSTALLED ON YOUR PHONE AND THIS IS THE FIRST TIME USING THIS TOOL
PLEASE UNINSTALL THE APP FIRST
Press any key to continue . . .
APK NOT FOUND. QUITTING.
‘sleep’ is not recognized as an internal or external command,
operable program or batch file.
C:\Users\super\AppData\Local\Temp\afolder\adb.exe
Access is denied.

**Comment by VWVideo** [comment]
The above steps worked for me BUT I can only play YouTube now. I've lost functionality to play local or external storage

Anyone else had this issue?


I'll have to fiddle around with plex i suppose...

**Comment by supercharger619** [comment]
I have a Chevy bolt 2017
Car stream-1.8.3
AA-I've tried both 3.0, and 2.9574923, currently have 2.9 loaded.

**Comment by masque3000** [comment]
Couldn't get it to work using AA3 + YTA183 in my 2018 Jeep Grand Cherokee with Uconnect. Didn't work with AA3 with any of the various methods (apk editor, patch, etc). I was gonna give up then I uninstalled both apps. I downloaded AA2.9 to my PC desktop and renamed it "my.apk". I used the patcher to install it. I opened the app and completed the setup on my phone, including developer options. Then I downloaded YTA183 to my PC desktop and used the patcher to install it. I am now watching videos in my car!

**Comment by tallpr24** [comment]
hey guys, when i run \Minimal ADB and Fastboot>adb shell pm install -i "com.android.vending" -r /Internal storage/Download/youtube_auto_v1.8.3.apk

it just sits there with a blinking cursor, it never completes..i have to eventually ctrl-c out of it, any ideas what i may be doing wrong? Im using galaxsy s9 plus and trying to get this to work on honda accord 2018 sport

**Comment by vascodgs** [comment]
I followed all the steps and it exactly like the video instructions but I plug my note 8 to my car it still doesn't work :/

**Comment by ahhl** [comment]
it seems that over at kiran blog, someone manage to get aa3.1 and AA patcher  to work and he having oreo

**Comment by sihfuh** [comment]
Hi.. Im trying to install CarStream through the adb.. but my phone (Samsung J7 Pro)'s internal paths name has space in it (for example: 'Internal storage' and 'SD card')...  It says No such file or directory..  and I cant rename both folders.. Need help please!...

Path: /SD card/APKS/CarStream_1.8.3.apk
Error failed to write; /SD no such file or directory

**Comment by marco69aurelio** [comment]
@sihfuh you may try to create a new folder without space in the name, then move the CarStream_1.8.3.apk there and re-try with ADB.

**Comment by joepk88** [comment]
Also I have aa2.9 patched and install from the pc we and then carstream patched and installed from the pc app posted earlier

**Comment by sihfuh** [comment]
@joepk88 Thanks.. I will try again with another explorer... and I will try to reinstall AA 2.9... I will let you guys know..

**Comment by vascodgs** [comment]
The problem is that Google is patching everything with Google play services updates 
What I used to do was downgrading to 2.9.57 and then installing carstream. Now the only way for the to work is by uninstalling Google play services updates on app settings. But as soon as you reinstall the updates, carstream is gone

**Comment by vascodgs** [comment]
I feel that the same is happening with 3.0 version 
And I tried this method with the new 3.1 version and it didn't work

**Comment by shmykelsa** [comment]
Hello there, I'm sorry for being radio silent these days. 

In a few words: Google just killed the switch.

I've been trying several method to get around it, and no easy method will do the trick right now. 

Altought my patcher is still pretty much functional, or at least it fully was on up to Android Auto 3.0, now Google is blocking server-side. There's someone claiming that rolling back to 2.9 is the solution, and CarStream is still working, but my suggestion is to don't rely so much on that. Eventually Google, through Google Play Services, will block the usage of custom apps on any version of AA, whether it's 2.8,2.9,3.0, you can rollback as much as you like, one way or another Google will bust you and force you to update and no patcher will do, and don't expect a new patcher right now.

So what now? My work here is basically over until a new solution is found, but I'm pretty positive no easy solution will make it right now. It's basically all down to @thekirankumar to try and solve it (I personally spoke to him, he's fine). 

For the records, I have raised a thread on Android Auto's product forum, and you can find it here https://productforums.google.com/forum/#!topic/android-auto/pLTqJpGGFPs;context-place=forum/android-auto but I reckon Google but I think that our opinion on this matter it's water off a duck's back, because YouTube Auto is what got Google angry on the first place.

**Comment by ibdagriff** [comment]
Watch the instructional video.  Installed everything correct.  I got a "success" at the command prompt. Go to run carstream and still no carstream link in the last tab.

**Comment by ibdagriff** [comment]
Watched the instructional video.  Installed everything correct.  I got a "success" at the command prompt. Go to run carstream and still no carstream link in the last tab.

**Comment by masque3000** [comment]
shmykelsa was absolutely right. I was using CarStream just fine on my way to the gym, but on my way home I saw an alert that 17 Play Store apps needed updating. I noticed "Instant Apps" was autoupdating although I disabled autoupdates and didn't know what that was. Googled it, disabled it, but CarStream disappeared & I haven't been able to use it or get it back since. Was great while it lasted!

**Comment by supercharger619** [comment]
Just lost functionality this morning, hopefully will hear something from
the app creater soon, there's got to be a work around for Android autos
incompatibility

**Comment by IAmScorpionz** [comment]
> 
Watched the instructional video. Installed everything correct. I got a "success" at the command prompt. Go to run carstream and still no carstream link in the last tab.

>
Me too. I already done all the instruction and also followed the step exactly like in the video but in the end still no carstream. 

The carstream only appear in the music tab. 

Running on Chevy MyLink

**Comment by kaisertgc** [comment]
Nothing should be impossible about programming. Somebody must find a solution? Yet? No? 🤨

**Comment by masque3000** [comment]
I have it working again, but only for so long. I uninstalled updates to Google Play Services. The drawback is I can't use Google Assistant (!), Google Pay, & Google App until I update Play Services. I found out when I tried to use "OK Google" to voice-text my husband on my way home from work. It spoke back saying something like "I'm sorry for the inconvenience---we have made improvements to such-and-such that require you to update Google Play Services.  Please unlock your device screen and choose accept to allow the updates."  I looked at my phone and it was asking me to update Play Services...

**Comment by shmykelsa** [comment]
Uninstalling or rolling back Google Play Services is just not worth it.

**Comment by masque3000** [comment]
I agree. I'll deal with it until I can't deal anymore. Didn't realize how much I depend on Play Svcs...
